iwyu.cc

```diff
@@ -3454,13 +3453,8 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     // When the alias is from template (like 'Type' in
     // 'Tpl<Providing>::Type'), the type may be provided with template
     // argument and should not be reported. 'ReportTypeUse' performs that
-    // analysis when passing 'ElaboratedType' into it.
-    if (const auto* elaborated =
-            current_ast_node()->template GetParentAs<ElaboratedType>()) {
-      ReportTypeUse(CurrentLoc(), elaborated, DerefKind::None);
-    } else {
-      ReportTypeUse(CurrentLoc(), type, DerefKind::None);
-    }
+    // analysis when passing elaborated type into it.
+    ReportTypeUse(CurrentLoc(), type, DerefKind::None);
   }
```

Comment:
`ReportWrittenTypeAlias` function just delegates now to `ReportTypeUse`. I'm not sure if it should be leaved here.